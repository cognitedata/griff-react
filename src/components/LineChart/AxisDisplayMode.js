const AxisDisplayMode = {
  ALL: {
    id: 'ALL',
    width: (axisWidth, numAxes) => +axisWidth * +numAxes,
    toString: () => 'ALL',
  },
  NONE: { id: 'NONE', width: () => 0, toString: () => 'NONE' },
  COLLAPSED: {
    id: 'COLLAPSED',
    width: (axisWidth, numAxes = 0) => Math.min(+numAxes, 1) * +axisWidth,
    toString: () => 'COLLAPSED',
  },
};

export default AxisDisplayMode;
