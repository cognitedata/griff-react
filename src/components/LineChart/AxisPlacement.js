const AxisPlacement = {
  UNSPECIFIED: { id: 0, name: 'UNSPECIFIED', toString: () => 'UNSPECIFIED' },
  RIGHT: { id: 1, name: 'RIGHT', toString: () => 'RIGHT' },
  LEFT: { id: 2, name: 'LEFT', toString: () => 'LEFT' },
  BOTH: { id: 3, name: 'BOTH', toString: () => 'BOTH' },
};

export default AxisPlacement;
