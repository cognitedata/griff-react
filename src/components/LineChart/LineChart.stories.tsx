import React from 'react';
import { sineLoader, StoryContainer } from 'storybook/utils';
import DataProvider from 'components/DataProvider';
import Series from 'components/Series';
import LineChart from '.';

const DEFAULT_HEIGHT = 500;

export default {
  title: 'Components|LineChart',
};

type StoryProps = {
  size?: { width: number; height: number };
};

export const Base = ({ size }: StoryProps) => {
  const timeDomain = [0, 10000000];
  return (
    <StoryContainer>
      <DataProvider defaultLoader={sineLoader} timeDomain={timeDomain}>
        <Series id="1" color="steelblue" />
        <LineChart height={DEFAULT_HEIGHT} size={size} />
      </DataProvider>
    </StoryContainer>
  );
};
