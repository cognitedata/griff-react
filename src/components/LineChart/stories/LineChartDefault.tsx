import React from 'react';
import LineChart from '..';
import { LineChartProps } from '../types';
import { DEFAULT_HEIGHT, DEFAULT_SIZE } from './constants';

const LineChartDefault = (props: Partial<LineChartProps>) => {
  return <LineChart height={DEFAULT_HEIGHT} size={DEFAULT_SIZE} {...props} />;
};

export default LineChartDefault;
