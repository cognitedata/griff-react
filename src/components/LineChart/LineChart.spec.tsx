import React from 'react';
import { render, waitForElement } from '@testing-library/react';
import { Base } from './LineChart.stories';

describe('LineChart', () => {
  it('should base render correctly', async () => {
    const { asFragment, queryByTestId } = render(<Base />);

    await waitForElement(() => queryByTestId('Line-1'));
    expect(asFragment()).toMatchSnapshot();
  });
});
