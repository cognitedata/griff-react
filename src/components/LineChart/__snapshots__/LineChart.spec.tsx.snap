// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LineChart should base render correctly 1`] = `
<DocumentFragment>
  <div
    style="display: flex; justify-content: center; padding: 2rem;"
  >
    <div
      style="width: 100%; max-width: 1000px;"
    >
      <div
        class="linechart-container"
        style="display: grid; grid-template-areas: 'chart yaxis' 'xaxis .' 'context .'; height: 100%;"
      >
        <div
          class="lines-container"
          style="grid-area: chart; display: flex; flex-direction: column; height: 100%;"
        >
          <svg
            height="300"
            width="950"
          >
            <g
              height="300"
              width="950"
            >
              <clippath
                id="clip-path-950-300-1-0-ALL"
              >
                <rect
                  fill="none"
                  height="300"
                  width="950"
                />
              </clippath>
              <g
                clip-path="url(#clip-path-950-300-1-0-ALL)"
              >
                <path
                  class="line"
                  d="M0,150L3.8000000000000003,146.40970289047067L7.6000000000000005,142.82167348954803L11.4,139.23817807350528L15.200000000000001,135.6614800548538L19,132.09383855272284L22.8,128.53750696595196L26.6,124.9947315497956L30.400000000000002,121.46774999714023L34.199999999999996,117.95879002512984L38,114.47006796809168L41.8,111.00378737765209L45.6,107.56213763092617L49.4,104.14729254766027L53.2,100.76140901720072L57,97.40662563615626L60.800000000000004,94.08506135761488L64.60000000000001,90.79881415276725L68.39999999999999,87.54995968578345L72.2,84.3405500027788L76,81.17261223569761L79.80000000000001,78.04814732193333L83.6,74.9691287404934L87.39999999999999,71.93750126550759L91.2,68.9551797378673L95,66.02404785577056L98.8,63.14595698493832L102.6,60.32272498925187L106.4,57.556135082551094L110.2,54.847934702318526L114,52.199834405960274L117.8,49.61350679038125L121.60000000000001,47.090585435537264L125.4,44.63266387263087L129.20000000000002,42.241294577602844L133,39.917987990554934L136.79999999999998,37.66421156172407L140.6,35.48138882460876L144.4,33.37089849683491L148.2,31.334073609328755L152,29.372200664345144L155.8,27.48651882288584L159.60000000000002,25.678219122018447L163.39999999999998,23.948443722592003L167.2,22.298285187823865L171,20.728785793212978L174.79999999999998,19.240936868216806L178.6,17.83567817010595L182.4,16.513897290393686L186.20000000000002,15.276429094213768L190,14.12405519300222L193.79999999999998,13.057503450814124L197.6,12.0774475245899L201.4,11.184506438658389L205.2,10.37924419374815L209,9.662169410752767L212.8,9.033735009474299L216.6,8.494337922550244L220.4,8.044318844741781L224.2,7.683962017743852L228,7.413495050652159L231.79999999999998,7.23308877620002L235.6,7.142857142857167L239.4,7.142857142857167L243.20000000000002,7.23308877620002L247,7.413495050652159L250.8,7.683962017743852L254.60000000000002,8.044318844741781L258.40000000000003,8.494337922550244L262.20000000000005,9.033735009474299L266,9.662169410752767L269.79999999999995,10.37924419374815L273.59999999999997,11.184506438658389L277.4,12.0774475245899L281.2,13.057503450814124L285,14.124055193002164L288.8,15.276429094213768L292.6,16.51389729039363L296.4,17.83567817010595L300.2,19.240936868216806L304,20.728785793213035L307.8,22.298285187823865L311.6,23.948443722591946L315.40000000000003,25.678219122018447L319.20000000000005,27.48651882288584L323,29.372200664345144L326.79999999999995,31.334073609328698L330.59999999999997,33.37089849683491L334.4,35.4813888246087L338.2,37.66421156172407L342,39.917987990554934L345.8,42.24129457760279L349.59999999999997,44.63266387263087L353.4,47.090585435537264L357.2,49.61350679038125L361,52.199834405960274L364.8,54.847934702318526L368.6,57.55613508255104L372.40000000000003,60.32272498925184L376.20000000000005,63.14595698493838L380,66.02404785577056L383.8,68.9551797378673L387.59999999999997,71.93750126550756L391.4,74.96912874049332L395.2,78.04814732193341L399,81.17261223569764L402.8,84.34055000277877L406.59999999999997,87.54995968578345L410.4,90.79881415276722L414.2,94.08506135761482L418,97.40662563615629L421.8,100.76140901720072L425.6,104.14729254766027L429.40000000000003,107.56213763092617L433.2,111.00378737765203L437,114.47006796809168L440.8,117.95879002512987L444.6,121.46774999714023L448.4,124.99473154979552L452.2,128.53750696595196L456,132.09383855272284L459.8,135.66148005485377L463.59999999999997,139.2381780735053L467.4,142.82167348954803L471.2,146.40970289047064L475,149.99999999999997L478.8,153.59029710952933L482.6,157.17832651045197L486.40000000000003,160.7618219264947L490.2,164.33851994514617L494,167.90616144727707L497.8,171.46249303404807L501.6,175.00526845020443L505.40000000000003,178.5322500028597L509.20000000000005,182.04120997487016L513,185.52993203190832L516.8000000000001,188.9962126223479L520.6,192.43786236907386L524.4000000000001,195.8527074523397L528.2,199.23859098279928L532,202.59337436384374L535.8,205.9149386423851L539.5999999999999,209.20118584723275L543.4,212.4500403142166L547.1999999999999,215.6594499972212L551,218.82738776430233L554.8,221.95185267806664L558.6,225.03087125950663L562.4,228.0624987344924L566.1999999999999,231.04482026213273L570,233.97595214422944L573.8,236.85404301506162L577.6,239.67727501074813L581.4,242.44386491744893L585.2,245.15206529768147L589,247.80016559403978L592.8,250.38649320961878L596.6,252.90941456446268L600.4,255.36733612736913L604.2,257.7587054223972L608,260.08201200944507L611.8000000000001,262.33578843827587L615.6,264.5186111753913L619.4,266.6291015031651L623.2,268.66592639067125L627,270.62779933565486L630.8000000000001,272.5134811771142L634.6,274.3217808779816L638.4000000000001,276.051556277408L642.2,277.70171481217614L646,279.27121420678696L649.8000000000001,280.7590631317832L653.5999999999999,282.16432182989405L657.4,283.4861027096063L661.1999999999999,284.72357090578623L665,285.87594480699784L668.8,286.9424965491858L672.5999999999999,287.9225524754101L676.4,288.81549356134167L680.1999999999999,289.6207558062518L684,290.3378305892473L687.8,290.9662649905257L691.6,291.50566207744976L695.4,291.9556811552582L699.1999999999999,292.3160379822561L703,292.58650494934784L706.8,292.7669112238L710.6,292.85714285714283L714.4,292.85714285714283L718.2,292.7669112238L722,292.58650494934784L725.8000000000001,292.3160379822561L729.6,291.9556811552582L733.4,291.50566207744976L737.2,290.9662649905257L741,290.3378305892473L744.8000000000001,289.62075580625185L748.6,288.81549356134167L752.4000000000001,287.9225524754101L756.2,286.9424965491858L760,285.87594480699784L763.8000000000001,284.72357090578623L767.6,283.4861027096063L771.4000000000001,282.16432182989405L775.1999999999999,280.7590631317832L779,279.271214206787L782.8,277.7017148121762L786.5999999999999,276.051556277408L790.4,274.32178087798155L794.1999999999999,272.5134811771142L798,270.62779933565486L801.8,268.6659263906713L805.6,266.62910150316515L809.4,264.5186111753913L813.1999999999999,262.3357884382759L817,260.08201200944507L820.8,257.7587054223972L824.6,255.36733612736913L828.4,252.90941456446282L832.2,250.38649320961875L836,247.8001655940397L839.8,245.1520652976815L843.6,242.44386491744893L847.4,239.6772750107481L851.2,236.8540430150617L855,233.97595214422947L858.8000000000001,231.04482026213273L862.6,228.0624987344925L866.4,225.03087125950668L870.2,221.9518526780667L874,218.8273877643024L877.8000000000001,215.6594499972212L881.6,212.4500403142165L885.4000000000001,209.2011858472328L889.2,205.91493864238515L893,202.59337436384374L896.8,199.2385909827994L900.5999999999999,195.85270745233976L904.4,192.43786236907386L908.1999999999999,188.99621262234803L912,185.5299320319084L915.8,182.04120997487007L919.6,178.53225000285977L923.4,175.00526845020443L927.1999999999999,171.46249303404798L931,167.90616144727719L934.8,164.33851994514623L938.6,160.7618219264947L942.4,157.17832651045208L946.2,153.5902971095294"
                  data-testid="Line-1"
                  style="stroke: steelblue; stroke-width: 1px; stroke-opacity: 1; fill: none; display: inherit;"
                />
              </g>
            </g>
            <rect
              fill="none"
              height="300"
              pointer-events="all"
              width="950"
            />
          </svg>
        </div>
        <div
          class="y-axis-container"
          style="grid-area: yaxis; display: flex; flex-direction: column;"
        >
          <svg
            height="300"
            width="50"
          >
            <g
              class="axis-y axis-y-1"
              cursor="move"
              transform="translate(0, 0)"
            >
              <g
                class="axis y-axis"
                fill="none"
                font-size="14"
                stroke-width="2"
                text-anchor="start"
              >
                <path
                  d="M6,299H1V1H6"
                  stroke="steelblue"
                />
                <g
                  opacity="1"
                  transform="translate(0, 292.86842314724197)"
                >
                  <line
                    stroke="steelblue"
                    x1="0"
                    x2="6"
                    y1="1"
                    y2="1"
                  />
                  <text
                    class="tick-value"
                    dy="0.32em"
                    fill="steelblue"
                    x="9"
                    y="1"
                  >
                    -1
                  </text>
                </g>
                <g
                  opacity="1"
                  transform="translate(0, 221.43421157362098)"
                >
                  <line
                    stroke="steelblue"
                    x1="0"
                    x2="6"
                    y1="1"
                    y2="1"
                  />
                  <text
                    class="tick-value"
                    dy="0.32em"
                    fill="steelblue"
                    x="9"
                    y="1"
                  >
                    -0.5
                  </text>
                </g>
                <g
                  opacity="1"
                  transform="translate(0, 150)"
                >
                  <line
                    stroke="steelblue"
                    x1="0"
                    x2="6"
                    y1="1"
                    y2="1"
                  />
                  <text
                    class="tick-value"
                    dy="0.32em"
                    fill="steelblue"
                    x="9"
                    y="1"
                  >
                    0
                  </text>
                </g>
                <g
                  opacity="1"
                  transform="translate(0, 78.56578842637902)"
                >
                  <line
                    stroke="steelblue"
                    x1="0"
                    x2="6"
                    y1="1"
                    y2="1"
                  />
                  <text
                    class="tick-value"
                    dy="0.32em"
                    fill="steelblue"
                    x="9"
                    y="1"
                  >
                    0.5
                  </text>
                </g>
                <g
                  opacity="1"
                  transform="translate(0, 7.1315768527579735)"
                >
                  <line
                    stroke="steelblue"
                    x1="0"
                    x2="6"
                    y1="1"
                    y2="1"
                  />
                  <text
                    class="tick-value"
                    dy="0.32em"
                    fill="steelblue"
                    x="9"
                    y="1"
                  >
                    1
                  </text>
                </g>
              </g>
              <rect
                fill="none"
                height="300"
                pointer-events="all"
                width="50"
              />
            </g>
          </svg>
        </div>
        <div
          class="x-axis-container"
          style="grid-area: xaxis; display: flex; flex-direction: row; width: 100%;"
        >
          <svg
            height="50"
            style="width: 100%; display: block;"
            width="950"
          >
            <g
              class="axis x-axis"
              data-testid="xAxis"
              fill="none"
              font-size="14"
              stroke-width="2"
              text-anchor="middle"
            >
              <path
                d="M1,6V1H949V6"
                stroke="black"
              />
              <g
                opacity="1"
                transform="translate(0, 0)"
              >
                <line
                  stroke="black"
                  x1="1"
                  x2="1"
                  y2="6"
                />
                <text
                  class="tick-value"
                  dy="0.71em"
                  fill="black"
                  x="1"
                  y="9"
                >
                  01:00
                </text>
              </g>
              <g
                opacity="1"
                transform="translate(85.5, 0)"
              >
                <line
                  stroke="black"
                  x1="1"
                  x2="1"
                  y2="6"
                />
                <text
                  class="tick-value"
                  dy="0.71em"
                  fill="black"
                  x="1"
                  y="9"
                >
                  01:15
                </text>
              </g>
              <g
                opacity="1"
                transform="translate(171, 0)"
              >
                <line
                  stroke="black"
                  x1="1"
                  x2="1"
                  y2="6"
                />
                <text
                  class="tick-value"
                  dy="0.71em"
                  fill="black"
                  x="1"
                  y="9"
                >
                  01:30
                </text>
              </g>
              <g
                opacity="1"
                transform="translate(256.5, 0)"
              >
                <line
                  stroke="black"
                  x1="1"
                  x2="1"
                  y2="6"
                />
                <text
                  class="tick-value"
                  dy="0.71em"
                  fill="black"
                  x="1"
                  y="9"
                >
                  01:45
                </text>
              </g>
              <g
                opacity="1"
                transform="translate(342, 0)"
              >
                <line
                  stroke="black"
                  x1="1"
                  x2="1"
                  y2="6"
                />
                <text
                  class="tick-value"
                  dy="0.71em"
                  fill="black"
                  x="1"
                  y="9"
                >
                  02:00
                </text>
              </g>
              <g
                opacity="1"
                transform="translate(427.5, 0)"
              >
                <line
                  stroke="black"
                  x1="1"
                  x2="1"
                  y2="6"
                />
                <text
                  class="tick-value"
                  dy="0.71em"
                  fill="black"
                  x="1"
                  y="9"
                >
                  02:15
                </text>
              </g>
              <g
                opacity="1"
                transform="translate(513, 0)"
              >
                <line
                  stroke="black"
                  x1="1"
                  x2="1"
                  y2="6"
                />
                <text
                  class="tick-value"
                  dy="0.71em"
                  fill="black"
                  x="1"
                  y="9"
                >
                  02:30
                </text>
              </g>
              <g
                opacity="1"
                transform="translate(598.5, 0)"
              >
                <line
                  stroke="black"
                  x1="1"
                  x2="1"
                  y2="6"
                />
                <text
                  class="tick-value"
                  dy="0.71em"
                  fill="black"
                  x="1"
                  y="9"
                >
                  02:45
                </text>
              </g>
              <g
                opacity="1"
                transform="translate(684, 0)"
              >
                <line
                  stroke="black"
                  x1="1"
                  x2="1"
                  y2="6"
                />
                <text
                  class="tick-value"
                  dy="0.71em"
                  fill="black"
                  x="1"
                  y="9"
                >
                  03:00
                </text>
              </g>
              <g
                opacity="1"
                transform="translate(769.5, 0)"
              >
                <line
                  stroke="black"
                  x1="1"
                  x2="1"
                  y2="6"
                />
                <text
                  class="tick-value"
                  dy="0.71em"
                  fill="black"
                  x="1"
                  y="9"
                >
                  03:15
                </text>
              </g>
              <g
                opacity="1"
                transform="translate(855, 0)"
              >
                <line
                  stroke="black"
                  x1="1"
                  x2="1"
                  y2="6"
                />
                <text
                  class="tick-value"
                  dy="0.71em"
                  fill="black"
                  x="1"
                  y="9"
                >
                  03:30
                </text>
              </g>
              <g
                opacity="1"
                transform="translate(940.5, 0)"
              >
                <line
                  stroke="black"
                  x1="1"
                  x2="1"
                  y2="6"
                />
                <text
                  class="tick-value"
                  dy="0.71em"
                  fill="black"
                  x="1"
                  y="9"
                >
                  03:45
                </text>
              </g>
            </g>
            <rect
              fill="none"
              height="50"
              pointer-events="all"
              width="950"
            />
            <div
              class="erd_scroll_detection_container erd_scroll_detection_container_animation_active"
              style="visibility: hidden; display: inline; width: 0px; height: 0px; z-index: -1; overflow: hidden; margin: 0px; padding: 0px;"
            >
              <div
                class="erd_scroll_detection_container"
                dir="ltr"
                style="position: absolute; flex: 0 0 auto; overflow: hidden; z-index: -1; visibility: hidden; width: 100%; height: 100%; left: 0px; top: 0px;"
              >
                <div
                  class="erd_scroll_detection_container"
                  style="position: absolute; flex: 0 0 auto; overflow: hidden; z-index: -1; visibility: hidden; left: -501px; top: -501px; right: -500px; bottom: -500px;"
                >
                  <div
                    style="position: absolute; flex: 0 0 auto; overflow: scroll; z-index: -1; visibility: hidden; width: 100%; height: 100%;"
                  >
                    <div
                      style="position: absolute; left: 0px; top: 0px;"
                    />
                  </div>
                  <div
                    style="position: absolute; flex: 0 0 auto; overflow: scroll; z-index: -1; visibility: hidden; width: 100%; height: 100%;"
                  >
                    <div
                      style="position: absolute; width: 200%; height: 200%;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </svg>
        </div>
        <div
          style="grid-area: spacer;"
        />
        <div
          class="context-container"
          style="grid-area: context; display: flex; flex-direction: column;"
        >
          <svg
            height="100"
            style="width: 100%; display: block;"
            width="950"
          >
            <g
              height="100"
              width="950"
            >
              <clippath
                id="clip-path-950-100-1-0-ALL"
              >
                <rect
                  fill="none"
                  height="100"
                  width="950"
                />
              </clippath>
              <g
                clip-path="url(#clip-path-950-100-1-0-ALL)"
              >
                <path
                  class="line"
                  d="M0,50L3.8000000000000003,48.80323429682356L7.6000000000000005,47.60722449651601L11.4,46.41272602450176L15.200000000000001,45.22049335161793L19,44.03127951757428L22.8,42.84583565531732L26.6,41.66491051659853L30.400000000000002,40.48924999904674L34.199999999999996,39.31959667504328L38,38.156689322697225L41.8,37.001262459217365L45.6,35.85404587697539L49.4,34.715764182553414L53.2,33.5871363390669L57,32.46887521205208L60.800000000000004,31.36168711920496L64.60000000000001,30.266271384255745L68.39999999999999,29.18331989526115L72.2,28.113516667592933L76,27.057537411899204L79.80000000000001,26.01604910731112L83.6,24.989709580164458L87.39999999999999,23.979167088502535L91.2,22.985059912622432L95,22.008015951923525L98.8,21.048652328312784L102.6,20.107574996417284L106.4,19.185378360850365L110.2,18.282644900772837L114,17.39994480198675L117.8,16.537835596793755L121.60000000000001,15.69686181184575L125.4,14.87755462421029L129.20000000000002,14.08043152586761L133,13.30599599685165L136.79999999999998,12.554737187241344L140.6,11.82712960820291L144.4,11.123632832278304L148.2,10.444691203109585L152,9.79073355478171L155.8,9.162172940961938L159.60000000000002,8.559406374006144L163.39999999999998,7.982814574197334L167.2,7.432761729274617L171,6.909595264404331L174.79999999999998,6.41364562273894L178.6,5.9452260567019835L182.4,5.504632430131224L186.20000000000002,5.092143031404589L190,4.708018397667402L193.79999999999998,4.3525011502713795L197.6,4.025815841529962L201.4,3.728168812886125L205.2,3.459748064582712L209,3.2207231369175844L212.8,3.011245003158109L216.6,2.831445974183424L220.4,2.681439614913927L224.2,2.5613206725812887L228,2.4711650168840436L231.79999999999998,2.4110295920666687L235.6,2.380952380952394L239.4,2.380952380952394L243.20000000000002,2.4110295920666687L247,2.4711650168840436L250.8,2.5613206725812887L254.60000000000002,2.681439614913927L258.40000000000003,2.831445974183424L262.20000000000005,3.011245003158109L266,3.2207231369175844L269.79999999999995,3.459748064582712L273.59999999999997,3.728168812886125L277.4,4.025815841529962L281.2,4.3525011502713795L285,4.708018397667374L288.8,5.092143031404589L292.6,5.504632430131224L296.4,5.9452260567019835L300.2,6.41364562273894L304,6.909595264404345L307.8,7.432761729274617L311.6,7.98281457419732L315.40000000000003,8.559406374006144L319.20000000000005,9.162172940961938L323,9.79073355478171L326.79999999999995,10.44469120310957L330.59999999999997,11.12363283227829L334.4,11.827129608202895L338.2,12.554737187241358L342,13.30599599685165L345.8,14.080431525867596L349.59999999999997,14.87755462421029L353.4,15.69686181184575L357.2,16.537835596793755L361,17.39994480198675L364.8,18.282644900772837L368.6,19.18537836085035L372.40000000000003,20.10757499641727L376.20000000000005,21.048652328312784L380,22.008015951923525L383.8,22.985059912622432L387.59999999999997,23.97916708850252L391.4,24.989709580164444L395.2,26.016049107311133L399,27.05753741189922L402.8,28.11351666759292L406.59999999999997,29.18331989526115L410.4,30.266271384255745L414.2,31.361687119204944L418,32.4688752120521L421.8,33.5871363390669L425.6,34.715764182553414L429.40000000000003,35.85404587697539L433.2,37.001262459217344L437,38.156689322697225L440.8,39.319596675043286L444.6,40.48924999904674L448.4,41.66491051659851L452.2,42.84583565531732L456,44.03127951757428L459.8,45.22049335161792L463.59999999999997,46.41272602450177L467.4,47.60722449651601L471.2,48.80323429682355L475,49.999999999999986L478.8,51.196765703176446L482.6,52.39277550348398L486.40000000000003,53.587273975498235L490.2,54.77950664838206L494,55.96872048242569L497.8,57.15416434468269L501.6,58.33508948340148L505.40000000000003,59.51075000095324L509.20000000000005,60.680403324956714L513,61.84331067730277L516.8000000000001,62.99873754078264L520.6,64.14595412302461L524.4000000000001,65.28423581744657L528.2,66.41286366093308L532,67.5311247879479L535.8,68.63831288079504L539.5999999999999,69.73372861574425L543.4,70.81668010473886L547.1999999999999,71.88648333240707L551,72.94246258810077L554.8,73.98395089268888L558.6,75.01029041983554L562.4,76.02083291149746L566.1999999999999,77.01494008737757L570,77.99198404807647L573.8,78.9513476716872L577.6,79.8924250035827L581.4,80.81462163914964L585.2,81.71735509922715L589,82.60005519801325L592.8,83.46216440320626L596.6,84.30313818815424L600.4,85.12244537578971L604.2,85.91956847413239L608,86.69400400314835L611.8000000000001,87.44526281275863L615.6,88.17287039179709L619.4,88.8763671677217L623.2,89.55530879689042L627,90.20926644521829L630.8000000000001,90.83782705903808L634.6,91.44059362599386L638.4000000000001,92.01718542580267L642.2,92.56723827072538L646,93.09040473559565L649.8000000000001,93.58635437726106L653.5999999999999,94.05477394329802L657.4,94.49536756986878L661.1999999999999,94.9078569685954L665,95.29198160233261L668.8,95.6474988497286L672.5999999999999,95.97418415847004L676.4,96.27183118711389L680.1999999999999,96.54025193541727L684,96.77927686308243L687.8,96.98875499684189L691.6,97.16855402581658L695.4,97.31856038508607L699.1999999999999,97.43867932741871L703,97.52883498311596L706.8,97.58897040793333L710.6,97.61904761904762L714.4,97.61904761904762L718.2,97.58897040793333L722,97.52883498311596L725.8000000000001,97.43867932741871L729.6,97.31856038508607L733.4,97.16855402581658L737.2,96.98875499684189L741,96.77927686308243L744.8000000000001,96.54025193541727L748.6,96.27183118711389L752.4000000000001,95.97418415847002L756.2,95.6474988497286L760,95.29198160233261L763.8000000000001,94.90785696859541L767.6,94.49536756986878L771.4000000000001,94.05477394329802L775.1999999999999,93.58635437726106L779,93.09040473559567L782.8,92.5672382707254L786.5999999999999,92.01718542580268L790.4,91.44059362599384L794.1999999999999,90.83782705903806L798,90.20926644521829L801.8,89.55530879689043L805.6,88.87636716772171L809.4,88.1728703917971L813.1999999999999,87.44526281275864L817,86.69400400314836L820.8,85.9195684741324L824.6,85.12244537578971L828.4,84.30313818815428L832.2,83.46216440320624L836,82.60005519801324L839.8,81.71735509922718L843.6,80.81462163914964L847.4,79.8924250035827L851.2,78.95134767168724L855,77.99198404807649L858.8000000000001,77.01494008737757L862.6,76.0208329114975L866.4,75.01029041983556L870.2,73.9839508926889L874,72.9424625881008L877.8000000000001,71.88648333240707L881.6,70.81668010473884L885.4000000000001,69.73372861574427L889.2,68.63831288079504L893,67.5311247879479L896.8,66.41286366093313L900.5999999999999,65.28423581744659L904.4,64.14595412302461L908.1999999999999,62.99873754078268L912,61.8433106773028L915.8,60.680403324956686L919.6,59.51075000095326L923.4,58.335089483401475L927.1999999999999,57.15416434468266L931,55.96872048242573L934.8,54.779506648382075L938.6,53.587273975498235L942.4,52.39277550348403L946.2,51.19676570317646"
                  data-testid="Line-1"
                  style="stroke: steelblue; stroke-width: 1px; stroke-opacity: 1; fill: none; display: inherit;"
                />
              </g>
            </g>
            <g
              fill="none"
              stroke="#777"
            >
              <rect
                class="before-selection"
                fill="#777"
                fill-opacity="0.3"
                height="100"
                shape-rendering="crispEdges"
                stroke="#fff"
                width="0"
                x="0"
                y="0"
              />
              <rect
                class="after-selection"
                fill="#777"
                fill-opacity="0.3"
                height="100"
                shape-rendering="crispEdges"
                stroke="#fff"
                width="0"
                x="950"
                y="0"
              />
              <rect
                class="overlay"
                cursor="crosshair"
                fill="none"
                height="100"
                pointer-events="all"
                width="950"
                x="0"
                y="0"
              />
              <rect
                class="selection"
                cursor="move"
                fill="none"
                fill-opacity="0.3"
                height="100"
                pointer-events="all"
                shape-rendering="crispEdges"
                width="950"
                x="0"
                y="0"
              />
              <path
                class="handle handle--west"
                d="M 0 0 V 100"
                stroke="#333"
                stroke-width="2"
              />
              <rect
                class="handle-target handle-target--west"
                cursor="ew-resize"
                fill="none"
                height="100"
                pointer-events="all"
                stroke="none"
                width="10"
                x="-5"
                y="0"
              />
              <path
                class="handle handle--east"
                d="M 950 0 V 100"
                stroke="#333"
                stroke-width="2"
              />
              <rect
                class="handle-target handle-target--east"
                cursor="ew-resize"
                fill="none"
                height="100"
                pointer-events="all"
                stroke="none"
                width="10"
                x="945"
                y="0"
              />
            </g>
            <div
              class="erd_scroll_detection_container erd_scroll_detection_container_animation_active"
              style="visibility: hidden; display: inline; width: 0px; height: 0px; z-index: -1; overflow: hidden; margin: 0px; padding: 0px;"
            >
              <div
                class="erd_scroll_detection_container"
                dir="ltr"
                style="position: absolute; flex: 0 0 auto; overflow: hidden; z-index: -1; visibility: hidden; width: 100%; height: 100%; left: 0px; top: 0px;"
              >
                <div
                  class="erd_scroll_detection_container"
                  style="position: absolute; flex: 0 0 auto; overflow: hidden; z-index: -1; visibility: hidden; left: -501px; top: -501px; right: -500px; bottom: -500px;"
                >
                  <div
                    style="position: absolute; flex: 0 0 auto; overflow: scroll; z-index: -1; visibility: hidden; width: 100%; height: 100%;"
                  >
                    <div
                      style="position: absolute; left: 0px; top: 0px;"
                    />
                  </div>
                  <div
                    style="position: absolute; flex: 0 0 auto; overflow: scroll; z-index: -1; visibility: hidden; width: 100%; height: 100%;"
                  >
                    <div
                      style="position: absolute; width: 200%; height: 200%;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </svg>
          <svg
            height="50"
            style="width: 100%; display: block;"
            width="0"
          >
            <g
              class="axis x-axis"
              data-testid="xAxis"
              fill="none"
              font-size="14"
              stroke-width="2"
              text-anchor="middle"
            >
              <path
                d="M1,6V1H-1V6"
                stroke="black"
              />
              <g
                opacity="1"
                transform="translate(0, 0)"
              >
                <line
                  stroke="black"
                  x1="1"
                  x2="1"
                  y2="6"
                />
                <text
                  class="tick-value"
                  dy="0.71em"
                  fill="black"
                  x="1"
                  y="9"
                >
                  03:00
                </text>
              </g>
            </g>
            <rect
              fill="none"
              height="50"
              pointer-events="all"
              width="0"
            />
            <div
              class="erd_scroll_detection_container erd_scroll_detection_container_animation_active"
              style="visibility: hidden; display: inline; width: 0px; height: 0px; z-index: -1; overflow: hidden; margin: 0px; padding: 0px;"
            >
              <div
                class="erd_scroll_detection_container"
                dir="ltr"
                style="position: absolute; flex: 0 0 auto; overflow: hidden; z-index: -1; visibility: hidden; width: 100%; height: 100%; left: 0px; top: 0px;"
              >
                <div
                  class="erd_scroll_detection_container"
                  style="position: absolute; flex: 0 0 auto; overflow: hidden; z-index: -1; visibility: hidden; left: -501px; top: -501px; right: -500px; bottom: -500px;"
                >
                  <div
                    style="position: absolute; flex: 0 0 auto; overflow: scroll; z-index: -1; visibility: hidden; width: 100%; height: 100%;"
                  >
                    <div
                      style="position: absolute; left: 0px; top: 0px;"
                    />
                  </div>
                  <div
                    style="position: absolute; flex: 0 0 auto; overflow: scroll; z-index: -1; visibility: hidden; width: 100%; height: 100%;"
                  >
                    <div
                      style="position: absolute; width: 200%; height: 200%;"
                    />
                  </div>
                </div>
              </div>
            </div>
          </svg>
        </div>
        <div
          class="erd_scroll_detection_container erd_scroll_detection_container_animation_active"
          style="visibility: hidden; display: inline; width: 0px; height: 0px; z-index: -1; overflow: hidden; margin: 0px; padding: 0px;"
        >
          <div
            class="erd_scroll_detection_container"
            dir="ltr"
            style="position: absolute; flex: 0 0 auto; overflow: hidden; z-index: -1; visibility: hidden; width: 100%; height: 100%; left: 0px; top: 0px;"
          >
            <div
              class="erd_scroll_detection_container"
              style="position: absolute; flex: 0 0 auto; overflow: hidden; z-index: -1; visibility: hidden; left: -501px; top: -501px; right: -500px; bottom: -500px;"
            >
              <div
                style="position: absolute; flex: 0 0 auto; overflow: scroll; z-index: -1; visibility: hidden; width: 100%; height: 100%;"
              >
                <div
                  style="position: absolute; left: 0px; top: 0px; width: 1011px; height: 1011px;"
                />
              </div>
              <div
                style="position: absolute; flex: 0 0 auto; overflow: scroll; z-index: -1; visibility: hidden; width: 100%; height: 100%;"
              >
                <div
                  style="position: absolute; width: 200%; height: 200%;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
